### PR TITLE
Simplify how we create URL's in the UI

### LINKED ISSUE #675: Review how we concatenate strings to create URLs
**Describe exactly what you would like to see in an upcoming release**

Ref https://github.com/cylc/cylc-ui/pull/627#discussion_r632532438

We have URLs in Cylc UI that are created dynamically, like the JupyterHub user service URL used to start a WebSocket connection.

In some cases, the URL is getting an extra `/`. Libraries and browsers are able to handle the double slashes, but we should fix it in case it causes troubles with other libraries or in our code.

Ideally a library or existing code to join it. The simpler the merrier.

**Additional context**

- [x] Does it work when deploying the UI/UIS to a path? https://github.com/cylc/cylc-ui/issues/258
- [x] Do we need the Python provided BASE URL? https://github.com/cylc/cylc-ui/issues/549
  * I don't think we need the BASE URL being passed from Python anymore, and it should be OK to trust that the URL used in the browser will be the same one used to access GraphQL via Apollo Client and Hub Via Axios

**Pull requests welcome!**

### PR DESCRIPTION
**Pull request by kinow**
Closes #675 

Remove existing code that relied on the Python backend code to provide
the Base URL. Now this value is always whatever the browser URL contains.

There is now a single createUrl function that is used in the application.
This removes the need to use Vuex.

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Appropriate tests are included (unit and/or functional).
- [x] Appropriate change log entry included.
- [x] No documentation update required.

### CODE REVIEW

**Comment by kinow** [comment]
Just need some good testing before it can be marked as ready for review :+1:

**Comment by kinow** [comment]
Tested with the UI+UIS running locally, all good. Then enabled the path `c.JupyterHub.bind_url = 'http://127.0.0.1:8000/cylc'` and tested again (had to log-in again) and everything worked fine too.

**Comment by kinow** [comment]
The Base URL provided by Python doesn't seem to be needed. That was my first fix for an issue that was reported. But in hindsight I think it was never necessary, and simply using the URL from the browser would be sufficient to build the GraphQL & Hub API URL's.

**Comment by kinow** [comment]
If approved & merged, I will raise an issue in UIS to [remove the base URL](https://github.com/cylc/cylc-uiserver/blob/4a32321f8199ba5d9d5102fccc91a8b7664ed432/cylc/uiserver/handlers.py#L114-L115) from the rendered template :+1:

**Review by hjoliver — APPROVED** [review]
Looks fine to me :+1:

**Comment by kinow** [comment]
Rebased.

File: src/utils/urls.js [hunk 1]
```diff
@@ -16,34 +16,54 @@
...
-  const baseUrl = `${window.location.protocol}//${window.location.hostname}${port}${window.location.pathname}`
-  if (baseUrl.endsWith('//')) {
-    return baseUrl.substring(0, baseUrl.length - 1)
-  } else if (baseUrl.endsWith('/')) {
-    return baseUrl
-  }
-  return `${baseUrl}/`
+function unslash (url) {
+  return url.replace(/([^:]\/)\/+/g, '$1')
 }
```
Comment (oliver-sanders) [inline src/utils/urls.js:28]: I'm not sure what this is doing?

```node
> function unslash (url) {
...    return url.replace(/([^:]\/)\/+/g, '$1')
... }
undefined
> unslash('http://abc.def/g/h/ijk/l')
'http://abc.def/g/h/ijk/l'
```

Comment (kinow) [reply src/utils/urls.js:28]: Ah, sorry, missing docs. Will add some tomorrow, but happy to rename it too. Wrote the function name without thinking too much about it, sorry.

```js
unslash('http://abc.def///g////h/ijk/l')
"http://abc.def/g/h/ijk/l"
```

Comment (oliver-sanders) [reply src/utils/urls.js:28]: Ah, thanks, that makes sense, could call it `noramlise` or something like that.

**Review by oliver-sanders — APPROVED** [review]
Thanks, much simpler.

**Pull request merged**